Firestore/core/test/firebase/firestore/local/leveldb_mutation_queue_test.cc

```diff
@@ -0,0 +1,188 @@
+#include "Firestore/core/src/firebase/firestore/auth/user.h"
+#include "Firestore/core/src/firebase/firestore/local/leveldb_key.h"
+#include "Firestore/core/src/firebase/firestore/local/leveldb_mutation_queue.h"
+#include "Firestore/core/src/firebase/firestore/local/leveldb_persistence.h"
+#include "Firestore/core/src/firebase/firestore/local/reference_set.h"
+#include "Firestore/core/src/firebase/firestore/nanopb/byte_string.h"
+#include "Firestore/core/src/firebase/firestore/nanopb/fields_array.h"
+#include "Firestore/core/src/firebase/firestore/nanopb/message.h"
+#include "Firestore/core/src/firebase/firestore/nanopb/reader.h"
+#include "Firestore/core/src/firebase/firestore/nanopb/writer.h"
+#include "Firestore/core/src/firebase/firestore/util/ordered_code.h"
+#include "Firestore/core/test/firebase/firestore/local/mutation_queue_test.h"
+#include "Firestore/core/test/firebase/firestore/local/persistence_testing.h"
+#include "Firestore/core/test/firebase/firestore/util/status_testing.h"
+#include "absl/strings/string_view.h"
+#include "gtest/gtest.h"
+#include "leveldb/db.h"
+
+namespace firebase {
+namespace firestore {
+namespace local {
+namespace {
+
+using auth::User;
+using leveldb::DB;
+using leveldb::Slice;
+using leveldb::Status;
+using leveldb::WriteOptions;
+using model::BatchId;
+using nanopb::ByteString;
+using nanopb::Message;
+using nanopb::StringReader;
+using nanopb::StringWriter;
+using util::OrderedCode;
+
+// A dummy mutation value, useful for testing code that's known to examine only
+// mutation keys.
+const char* kDummy = "1";
+
+std::unique_ptr<Persistence> PersistenceFactory() {
+  return LevelDbPersistenceForTesting();
+}
+
+}  // namespace
+
+INSTANTIATE_TEST_CASE_P(LevelDbMutationQueueTest,
+                        MutationQueueTest,
+                        testing::Values(PersistenceFactory));
+
+class LevelDbMutationQueueTest : public MutationQueueTestBase {
+ public:
+  LevelDbMutationQueueTest()
+      : MutationQueueTestBase(PersistenceFactory()),
+        db_(static_cast<LevelDbPersistence*>(persistence_.get())->ptr()) {
+  }
+
+ protected:
+  void SetDummyValueForKey(const std::string& key);
+
+  DB* db_;
+};
+
+/**
+ * Creates a key that's structurally the same as LevelDbMutationKey except it
+ * allows for nonstandard table names.
+ */
+std::string MutationLikeKey(absl::string_view table,
+                            absl::string_view user_id,
+                            BatchId batch_id) {
+  std::string key;
+  OrderedCode::WriteSignedNumIncreasing(&key, 5);  // TableName
+  OrderedCode::WriteString(&key, table);
+  OrderedCode::WriteSignedNumIncreasing(&key, 13);  // UserId
+  OrderedCode::WriteString(&key, user_id);
+  OrderedCode::WriteSignedNumIncreasing(&key, 10);  // BatchId
+  OrderedCode::WriteSignedNumIncreasing(&key, batch_id);
+  OrderedCode::WriteSignedNumIncreasing(&key, 0);  // Terminator
+  return key;
+}
+
+TEST_F(LevelDbMutationQueueTest, LoadNextBatchID_zeroWhenTotallyEmpty) {
```

Comment:
Question: should `zero` be capitalized? Applies throughout this function.

Reply:
I removed all of these underscore separators (and then made the capitalization match).